File: dev-docs/snippets/groups/list.swift
import JsBaoClient

// List groups, optionally filtered by type. Returns a typed paginated envelope.
func list(client: JsBaoClient) async throws {
  // #region example
  let page = try await client.groups.list(options: ListGroupsOptions(type: "team", limit: 20))
  for group in page.items {
    print(group.groupId, group.name, group.memberCount)
  }
  let nextCursor = page.cursor
  // #endregion example
  _ = nextCursor
}

File: dev-docs/snippets/groups/list.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List groups, optionally filtered by type. Returns a paginated envelope.
export async function list(client: JsBaoClient) {
  // #region example
  const page = await client.groups.list({ type: "team", limit: 20 });
  for (const group of page.items) {
    console.log(group.groupId, group.name, group.memberCount);
  }
  const nextCursor = page.cursor;
  // #endregion example
  return nextCursor;
}
